File: SEGUNDO/C2/PDOO/backup/DeepSpace_ruby/lib/test_p1.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#encoding:utf-8

require_relative 'combat_result'
require_relative 'dice'
require_relative 'game_character'
require_relative 'loot'
require_relative 'shield_booster'
require_relative 'shot_result'
require_relative 'supplies_package'
require_relative 'weapon'
require_relative 'weapon_type'

  
class TestP1
  
  def self.main
    i = 1
    #Prueba de loot
    while i <= 5
      botin = Deepspace::Loot.new(i,i*2, i*3, i*4, i*10)
      puts "El valor de suplementos es ", botin.nSupplies
      puts "El valor de las armas es ", botin.nWeapons
      puts "El valor de los escudos es ", botin.nShields
      puts "El valor de los hangares es ", botin.nHangars
      puts "El valor de las medallas es ", botin.nMedals 

      i+= 1

      print "\n\n"
    end

    #Prueba de SuppliesPackage
    i = 1
    while i <= 5
      paquete=Deepspace::SuppliesPackage.new(i*100,i*1000,i*345)

      puts "El poder de municion es ", paquete.ammoPower
      puts "Las unidades de combustible son ", paquete.fuelUnits
      puts "El poder del escudo es ", paquete.shieldPower

      copia=Deepspace::SuppliesPackage.newCopy(paquete)
      puts "El poder de municion copiado es ", copia.ammoPower
      puts "Las unidades de combustible copiadas son ", copia.fuelUnits
      puts "El poder del escudo copiado es ", copia.shieldPower

      i+=1
      print "\n\n"
    end

    #Prueba de ShieldBooster
    i = 1
    while i <= 5
      potenciador_escudo=Deepspace::ShieldBooster.new("Escudo_de_obsidiana", i*10.0,i*i)

      puts "El valor de la mejora es ", potenciador_escudo.boost
      puts "Los usos restantes del escudo son ", potenciador_escudo.uses

      10.times {puts "La salida de useIt es ", potenciador_escudo.useIt, "Los usos restantes son ", potenciador_escudo.uses  }

      print "\n\n"

      copia=Deepspace::ShieldBooster.newCopy(potenciador_escudo)
      puts "El valor de la mejora es ", copia.boost
      puts "Los usos restantes del escudo son ", copia.uses

      10.times {puts "La salida de useIt copia es ", copia.useIt, "Los usos restantes de la copia son ", copia.uses  }

      i+=1

      print "\n\n"
    end

    #Prueba de Weapon
    i = 1
    while i <= 5
      arma1 = Deepspace::Weapon.new("Canon_de_protones",Deepspace::WeaponType::LASER, i)
      arma2 = Deepspace::Weapon.new("Canon_de_adrones", Deepspace::WeaponType::MISSILE,i*i)
      arma3 = Deepspace::Weapon.new("Arco_y_flecha", Deepspace::WeaponType::PLASMA, i*i*i)

      if(arma1.type == Deepspace::WeaponType::LASER)
        puts "El arma es LASER"
      end

      if(arma1.type == Deepspace::WeaponType::MISSILE)
        puts "El arma es MISSILE"
      end

      if(arma1.type == Deepspace::WeaponType::PLASMA)
        puts "El arma es PLASMA"
      end

      puts "Los usos restantes del arma 1 son " + arma1.uses.to_s
      puts "El poder del arma 1 es " + arma1.power.to_s
      10.times {puts "La salida de useIt en el arma 1 es " + arma1.useIt.to_s, "Los usos restantes son " + arma1.uses.to_s}

      print "\n\n"
      if(arma2.type == Deepspace::WeaponType::LASER)
        puts "El arma es LASER"
      end

      if(arma2.type == Deepspace::WeaponType::MISSILE)
        puts "El arma es MISSILE"
      end

      if(arma2.type == Deepspace::WeaponType::PLASMA)
        puts "El arma es PLASMA"
      end

      puts "Los usos restantes del arma 2 son " + arma2.uses.to_s
      puts "El poder del arma 2 es " + arma2.power.to_s
      10.times {puts "La salida de useIt en el arma 2 es " + arma2.useIt.to_s, "Los usos restantes son " + arma2.uses.to_s}

      print "\n\n"
      if(arma3.type == Deepspace::WeaponType::LASER)
        puts "El arma es LASER"
      end

      if(arma3.type == Deepspace::WeaponType::MISSILE)
        puts "El arma es MISSILE"
      end

      if(arma3.type == Deepspace::WeaponType::PLASMA)
        puts "El arma es PLASMA"
      end
      puts "Los usos restantes del arma 3 son " + arma3.uses.to_s
      puts "El poder del arma 3 es " + arma3.power.to_s
      10.times {puts "La salida de useIt en el arma 3 es " + arma3.useIt.to_s , "Los usos restantes son " + arma3.uses.to_s}

      i+=1

      print "\n\n"
    end

    #Prueba de Dice
    i=1
    nhangars=0
    nweapons=0
    nweapons2=0
    nweapons3=0
    nshields=0
    firstshot=0
    while i <= 100
      puts "Iniciando partida " + i.to_s
      jugador=rand(1..100)
      puts "Usted es el jugador " + jugador.to_s

      dado=Deepspace::Dice.new

      valorh=dado.initWithNHangars
      valorw=dado.initWithNWeapons
      valors=dado.initWithNShields

      if(valorh==0)
        nhangars += 1
      end

      if(valorw == 1)
        nweapons += 1

      elsif(valorw == 2)
        nweapons2 += 1

      else 
        nweapons3 += 1
      end




      if(valors == 0)
        nshields += 1
      end  

      puts "Numero de hangares asignado " + valorh.to_s
      puts "Numero de armas asignado " + valorw.to_s
      puts "Numero de escudos asignado " + valors.to_s

      puts "Empieza la partida el jugador " + dado.whoStarts(100).to_s

      if(dado.firstShot == Deepspace::GameCharacter::SPACESTATION)
        puts "El primer disparo lo realiza SPACESTATION"
        firstshot+=1

      else
        puts "El primer disparo lo realiza ENEMY STARSHIP"
      end

      velocidad=rand

      puts "La velocidad de su nave es ", velocidad
      if(dado.spaceStationMoves(velocidad))
        puts "Su nave ha esquivado el ataque de una nave enemiga"
      else
        puts "Su nave ha recibido un disparo y pierde una unidad de escudo"
      end

      i+=1

      print "\n\n"
    end

    puts "Estadisiticas"
    puts "Probabilidad de Hangar=0 ", nhangars/100.0
    puts "Probabilidad de Weapons=1", nweapons/100.0
    puts "Probabilidad de Weapons=2", nweapons2/100.0
    puts "Probabilidad de Weapons=3", nweapons3/100.0
    puts "Probabilidad de Shields=0", nshields/100.0
    puts "Probabilidad de FirstShot=SPACESTATION", firstshot/100.0
  
  end
  
  TestP1.main
end

File: SEGUNDO/C1/NetBeansProjects/Deep_space/src/deepspace/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;
import java.util.Random;
/**
 *
 * @author daniel
 */
public class TestP1 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
        
        Loot botin;
        SuppliesPackage paquete;
        ShieldBooster potenciador;
        Weapon arma1, arma2, arma3;
        Dice dado;
        int i=1;
        
        //Prueba de LOOT
        while(i <= 5){
            botin = new Loot(i,i*2,i*i,i*4,i*7);
            
            System.out.println("Cantidad de suplementos: " +botin.getNSupplies() + "\n");
            System.out.println("Cantidad de armas: " + botin.getNWeapons() + "\n");
            System.out.println("Cantidad de escudos: " + botin.getNShields() + "\n");
            System.out.println("Cantidad de hangares: " + botin.getNHangars() + "\n");
            System.out.println("Cantidad de medallas: " + botin.getNMedals() + "\n");
            
            i++;
        }
        
        i=1;
        
        //Prueba de SUPPLIESPACKAGE
        while(i <= 5){
            paquete=new SuppliesPackage(i*100,i*1000, i*500);
            
            System.out.println("Poder de armamento: " + paquete.getAmmoPower() + "\n");
            System.out.println("Unidades de combustible: " + paquete.getFuelUnits() + "\n");
            System.out.println("Poder de escudo: " + paquete.getShieldPower() + "\n");
            
            SuppliesPackage copia = new SuppliesPackage(paquete);
            
            System.out.println("Poder de armamento copiado: " + copia.getAmmoPower() + "\n");
            System.out.println("Unidades de combustible copiadas: " + copia.getFuelUnits() + "\n");
            System.out.println("Poder de escudo copiado: " + copia.getShieldPower() + "\n");
            
            i++;  
        }
        
        i=1;
        
        //Prueba SHIELDBOOSTER
        
        while(i <= 5){
            potenciador=new ShieldBooster("Escudo de hadrones", i*10.0f, i);
            
            System.out.println("El nombre del potenciador es " + potenciador.getName() + "\n");
            System.out.println("El poder del potenciador es " + potenciador.getBoost() + "\n");
            System.out.println("Los usos restantes del potenciador son " + potenciador.getUses() + "\n");
            
            for(int j=0; j < 7; j++)
                System.out.println("La salida de useIt es: " + potenciador.useIt() + "\n Los usos restantes son " + potenciador.getUses() + "\n");
            
            ShieldBooster copia = new ShieldBooster(potenciador);
            
            System.out.println("El nombre del potenciador copiado es " + copia.getName() + "\n");
            System.out.println("El poder del potenciador copiado es " + copia.getBoost() + "\n");
            System.out.println("Los usos restantes del potenciador copiado son " + copia.getUses() + "\n");
            
            for(int j=0; j < 7; j++)
                System.out.println("La salida de useIt es: " + potenciador.useIt() + "\n Los usos restantes son " + potenciador.getUses() + "\n");
            
            i++;
        }
        
        i=1;
        
        //Prueba WEAPON
        
        while (i <= 5){
            arma1= new Weapon("Arco_flecha", WeaponType.LASER, 10);
            arma2= new Weapon("Lanzamisiles", WeaponType.MISSILE, 4);
            arma3= new Weapon("Tanqueta_verde", WeaponType.PLASMA, 7);
            
            System.out.println("El arma 1 se llama: \n");
            
            if(arma1.getType()==WeaponType.LASER){
                System.out.println("EL arma 1 es de tipo laser\n");
            }
            
            else if (arma1.getType() == WeaponType.MISSILE)
                System.out.println("El arma 1 es de tipo misil");
            
            else
                System.out.println("El arma 1 es de tipo plasma");
            
            System.out.println("Al rama uno le quedan " + arma1.getUses() + "\n");
            
            for(int j=0; j <=10; j++){
                System.out.println("La salida de useIt en el arma 1 es: " + arma1.useIt() + "\n Los usos restantes del arma 1 son " + arma1.getUses() + "\n");
            }
            
            System.out.println("El arma 2 se llama: \n");
            
            if(arma2.getType()==WeaponType.LASER){
                System.out.println("EL arma 2 es de tipo laser\n");
            }
            
            else if (arma2.getType() == WeaponType.MISSILE)
                System.out.println("El arma 2 es de tipo misil");
            
            else
                System.out.println("El arma 2 es de tipo plasma");
            
            System.out.println("Al rama dos le quedan " + arma2.getUses() + "\n");
            
            for(int j=0; j <=10; j++){
                System.out.println("La salida de useIt en el arma 2 es: " + arma2.useIt() + "\n Los usos restantes del arma 2 son " + arma2.getUses() + "\n");
            }
            
            System.out.println("El arma 3 se llama: \n");
            
            if(arma3.getType()==WeaponType.LASER){
                System.out.println("EL arma 3 es de tipo laser\n");
            }
            
            else if (arma3.getType() == WeaponType.MISSILE)
                System.out.println("El arma 3 es de tipo misil");
            
            else
                System.out.println("El arma 3 es de tipo plasma");
            
            System.out.println("Al rama tres le quedan " + arma3.getUses() + "\n");
            
            for(int j=0; j <=10; j++){
                System.out.println("La salida de useIt en el arma 3 es: " + arma3.useIt() + "\n Los usos restantes del arma 3 son " + arma3.getUses() + "\n");
            }
            
            i++;            
        }
        
        i=1;
        
        int nhangars=0;
        int nweapons1=0;
        int nweapons2=0;
        int nweapons3=0;
        int nshields=0;
        int firstshot=0;
        //Prueba DICE
        
        while (i <= 100){
            System.out.println("Inicidando partida " + i +"\n");
            Random jugador=new Random();
            System.out.println("Se le ha asignado el jugador " + jugador.nextInt(100) + "\n");
            dado=new Dice();
            
            if(dado.initWithNHangar() == 0){
                System.out.println("Se le han asignado 0 hangares\n");
                nhangars++;
            }
            
            else
                System.out.println("Se le han asignado 1 hangar \n");
            
            int arma=dado.initWithWeapons();
            
            if(arma == 1){
                System.out.println("Se le ha asignado 1 arma\n");
                nweapons1++;
            }
            
            else if(arma == 2){
                System.out.println("Se le han asignado 2 armas\n");
                nweapons2++;
            }
            
            else{
                System.out.println("Se le han asignado 3 arma\n");
                nweapons3++;
            }
            
            if(dado.initWithNShields() == 0){
                System.out.println("No se le ha asignado escudo\n");
                nshields++;
            }
            
            else
                System.out.println("Se le ha asignado 1 escudo\n");
            
            
            System.out.println("La partida la comenzará el jugador: "+ dado.whoStarts(100) + "\n");
            
            
            if(dado.firstShot() == GameCharacter.SPACESTATION){
                System.out.println("El primer disparo lo realiza SPACESTATION");
                firstshot++;
            }
            
            else
                System.out.println("El primer disparo lo realiza ENEMYSTARSHIP");
            
            if(dado.spaceStationMoves(jugador.nextFloat()))
                System.out.println("Su nave ha esquivado el disparo\n");
            
            else
                System.out.println("Su nave ha recibido el disparo\n");
            
            i++;
            
            System.out.println("\n\n");
        }
        
        
        System.out.println("Estadísticas: \n Probabilidad hangar=0: " + nhangars / 100.0f + "\n Probabilidad arma=1: " +nweapons1/100.0f+ "\n Probabilidad arma=2: "+ nweapons2/100.0f +"\n Probabilidad arma=3: "+ nweapons3/100.0 + "\n Probabilidad firstshot=SPACESTATION: " + firstshot/100.0f + "\n");      
    }
    
}
